R/zchunk_L231.proc_sector.R

```diff
@@ -0,0 +1,460 @@
+    # Interpolate to specified historical years, as necessary
+    L231.rsrc_info <- A31.rsrc_info %>%
+      select(-year, -value) %>%
+      distinct() %>%
+      # Interpolate to all years
+      repeat_add_columns(tibble(year = HISTORICAL_YEARS)) %>%
+      left_join(A31.rsrc_info, by = c("resource", "resource_type",
+                                      "market", "output-unit", "price-unit", "capacity.factor", "year")) %>%
+      mutate(value = approx_fun(year, value, rule = 1)) %>%
+      filter(year %in% MODEL_YEARS) %>%
+      repeat_add_columns(GCAM_region_names) %>%
+      mutate(market = replace(market, market == "regional", region[market == "regional"]))
+
+    # L231.UnlimitRsrc: output unit, price unit, and market for unlimited resources
+    L231.UnlimitRsrc <- L231.rsrc_info %>%
+      filter(resource_type == "unlimited-resource") %>%
+      select(-year, - value) %>%
+      distinct %>%
+      select(region, unlimited.resource = resource, output.unit = `output-unit`, price.unit = `price-unit`, market, capacity.factor)
+
+    # L231.UnlimitRsrcPrice: prices for unlimited resources
+    L231.UnlimitRsrcPrice <- L231.rsrc_info %>%
+      filter(resource_type == "unlimited-resource") %>%
+      select(region, unlimited.resource = resource, year, price = value)
+
+    # L231.IndCoef: coefficient on industrial processes as an input to the industry sector
+    # Coefficient = 0.008 / change in industry output from 1990 (0.008 is the sum of calvalue)
+    # First, interpolate A32.globaltech_eff efficiency values to all years
+    Ind.globaltech_eff <- A32.globaltech_eff %>%
+      select(-year, -value) %>%
+      repeat_add_columns(tibble(year = c(HISTORICAL_YEARS, FUTURE_YEARS))) %>%
+      left_join(A32.globaltech_eff, by = c("supplysector", "subsector", "technology", "minicam.energy.input",
+                                           "secondary.output", "year")) %>%
+      group_by(supplysector, subsector, technology, minicam.energy.input,
+               secondary.output) %>%
+      mutate(value = round(approx_fun(year, value, rule = 1), 3)) %>%
+      ungroup %>%
+      filter(year %in% MODEL_YEARS) %>%
+      select(sector = supplysector, fuel = subsector, year, efficiency = value) %>%
+      distinct()
+
+    # Now combine input energy info and join with efficiency values
+    L231.IndCoef <- bind_rows(L1322.in_EJ_R_indenergy_F_Yh %>%
+                                       mutate(sector = "industrial energy use"),
+                                     L1322.in_EJ_R_indfeed_F_Yh %>%
+                                       mutate(sector = "industrial feedstocks")) %>%
+      left_join_keep_first_only(Ind.globaltech_eff, by = c("sector", "fuel", "year")) %>%
+      # Calculate service as energy * efficiency
+      mutate(service = value * efficiency) %>%
+      na.omit() %>%
+      group_by(GCAM_region_ID, year) %>%
+      summarise(ind_output = sum(service)) %>%
+      ungroup() %>%
+      mutate(ind_proc_input = 0.008,
```

Comment:
@bpbond there are a lot of hard cores #s in this file.  Do you know if they are in multiple files and would best to be pulled out in the constants?

Reply:
I'd suggest open an issue ("Lots of hardcoded numbers in L231.proc_sector") to remind us to follow up on this later.